go/vt/sqlannotations/sqlannotations.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package sqlannotations provides functions
+// for annotating DML statements with keyspace-id
+// comments and parsing them. These annotations
+// are used during filtered-replication to route
+// the DML statement to the correct shard.
+package sqlannotations
+
+import (
+	"encoding/hex"
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// AnnotateKeyspaceIdIfDML annotates 'sql' with the given keyspace id
+// if 'sql' is a DML statement; otherwise, it does nothing.
+func AnnotateKeyspaceIdIfDML(keyspaceId []byte, sql *string) {
```

Comment:
From outside this package, this always has to be called to as `sqlannotation.AnnotateKeyspaceIdIfDML`. In idiomatic Go, we try to avoid this stuttering (`annotation.Annotate`). Since the package name cannot be omitted, it's always clear already that the function is talking about a SQL annotation.

I suggest:
```
func AddKeyspaceID(sql string, keyspaceID []byte) string
```

This would be called as:
```
sql = sqlannotation.AddKeyspaceID(sql, keyspaceID)
```

My reasoning is:
* By returning the string instead of passing a pointer, it's more clear when the call site is modifying a value - and it becomes possible to use the function when you don't want to modify the original value.
* The reassignment and parameter order resembles an `append()` call, which is a pattern Go users are familiar with.
* By Go naming conventions, the `ID` in `KeyspaceID` should be all-caps. We have some legacy names that are slow to change since they are used in many places, but new names should follow the convention.
* I'm torn on the `IfDML` suffix, but I'd lean towards leaving it out and merely explaining in the docstring (as you did already) that it does nothing for non-DMLs.